File: Litchcode/dp/2025-redistribute-characters-to-make-all-strings-equal/redistribute-characters-to-make-all-strings-equal.go
func makeEqual(words []string) bool {
    counts := make(map[rune]int)
    for _,word:= range(words){
        for _,c := range(word){
            counts[c]++
        }
    }

    for _, count := range counts {
        if count%len(words) != 0 {
            return false
        }
    }
    return true
}

File: Litchcode/dp/2025-redistribute-characters-to-make-all-strings-equal/redistribute-characters-to-make-all-strings-equal.rs
impl Solution {
    pub fn make_equal(words: Vec<String>) -> bool {
        let mut alph = vec![0; 26];

        for word in &words {
            for c in word.chars() {
                alph[(c as u8 - b'a') as usize] += 1;
            }
        }
        alph.iter().all(|&count| count % words.len() == 0)
    }
}